### PR TITLE
A user should see special messages in digest when unusual cases

### PR DESCRIPTION
**Pull request by anatoliikryvishyn**

### CODE REVIEW

**Comment by serhiir** [comment]
@anatoliikryvishyn Btw, do you think about the 4th case (nothing done + everything is completed) if no data have been changed in the system during a few days in a row. Wouldn't it be annoying to send digest with "empty results" (there was nothing to do today) again and again?

**Comment by anatoliikryvishyn** [comment]
@serhiir, maybe when such situation appears - inform user one time, and alert him, that he will be notified when situation will change.

**Comment by anatoliikryvishyn** [comment]
@serhiir, please test [here](https://trouble-tada-interlink.herokuapp.com/) and review.
For sending mail you can use this [link](https://trouble-tada-interlink.herokuapp.com/items/statistic).
All items were deleted, so you can test case when nothing to do.

**Comment by serhiir** [comment]
@anatoliikryvishyn I'm not able to send a digest:

run error

File: app/views/daily_digest_mailer/_total_statistic.html.haml [hunk 1]
```diff
@@ -1,10 +1,22 @@
...
+        %p{style: 'font-size: 25px; display:inline-block; width:192px; margin-right: 26px; color: #34ad00;'}= t('digest.nothing_was_done')
+    - else
+      %td
+        %p{style: 'display:inline; color: #34ad00;'}= @statistic.completed_total_amount
+      %td
+        %p{style: 'font-size: 25px; display:inline-block; width:1px; margin-right: 150px; color: #34ad00;'}= t('digest.todos_completed')
+    - if @statistic.everything_is_done?
+      %td
+        %p{style: 'font-size: 25px; display:inline-block; width:160px; margin-right:12px; color: #1f8cd3;'}= t('digest.everything_is_done')
+    - else
```
Comment (serhiir) [inline app/views/daily_digest_mailer/_total_statistic.html.haml]: @anatoliikryvishyn I suggest extract appropriate pieces of the layout into partials as logic is mixed with the layout and it's hard to understand what is going on.

**Comment by serhiir** [comment]
@anatoliikryvishyn Done.

**Comment by anatoliikryvishyn** [comment]
@serhiir, now mail sending is working. Retest please.

**Comment by serhiir** [comment]
@anatoliikryvishyn Please take a look:

wrong statistics

The item was just created and then marked as completed.

**Comment by serhiir** [comment]
@anatoliikryvishyn Take a look at a similar bug: 

wrong text

**Comment by serhiir** [comment]
@anatoliikryvishyn Verified.

**Comment by anatoliikryvishyn** [comment]
@serhiir, updated, please review and retest. Bugs were fixed with simple server restart.

**Comment by serhiir** [comment]
@anatoliikryvishyn Ok, have you tried to retest yourself?

**Comment by anatoliikryvishyn** [comment]
@serhiir, yes I did with sending mails to myself.

**Comment by serhiir** [comment]
OK, thanks will take a look soon.

**Comment by serhiir** [comment]
Hey @anatoliikryvishyn, the applications sends two mails in a row instead of the one.

double mails

**Comment by serhiir** [comment]
@anatoliikryvishyn I'm confused, as sometimes it sends 2 copies and sometime one.

**Comment by serhiir** [comment]
@anatoliikryvishyn Done.

**Comment by anatoliikryvishyn** [comment]
@serhiir, strange situation. Tried to reproduce it, but still got 1 letter from one send request.
The only time that I received more then one letter it was, when sending few requests very fast ( e.g. two times pressed 'enter' key on [link](https://trouble-tada-interlink.herokuapp.com/items/statistic)). As in production there will be no such possibility to send mails through link ( it was made just for easy testing ) - this will not happen. But if you could tell me, please, how to reproduce this situation - then it will helps us a lot.

**Comment by serhiir** [comment]
@anatoliikryvishyn I just copied the link, pasted into the browser and pressed Enter.

**Comment by anatoliikryvishyn** [comment]
@serhiir, we found that Chrome has added prerendering for faster page loading. And so when user writes url for sending mail, but not go on it - chrome in background still managed this url and sends mail. That why we receive 2 mails ( one from background prerendering and one from 'enter' press ).
So to use this way of sending mails in Chrome you need to disable prerendering option in settings.

**Comment by anatoliikryvishyn** [comment]
Sergey, Is priority for next issues the same?

   1. completed items digest
   <https://github.com/anatoliikryvishyn/TaDa/issues/27>
   2. new lists digest <https://github.com/anatoliikryvishyn/TaDa/issues/28>


_________

Best regards,
Anatolii Kryvishyn
[image: phone:]   +380 938 789 329

[image: skype:]   kryvishyn.anatolii
[image: position:]   Web Developer

[image: InterLink]
InterLink LLC
Part of SP Group
[image: phone:]   +380 47 233 05 88

[image: location:]   Ukraine, Lviv, 79000
[image: site:]   interlink-ua.com




This email and any files transmitted with it are confidential and intended
solely for the use of the individual or entity to whom they are addressed.
If you have received this email in error please notify the sender
immediately.

**Comment by serhiir** [comment]
@anatoliikryvishyn Yes, please.

**Pull request merged**